### ISSUE TITLE
Name mismatch when generating `Show` instances for custom promoted types

### ISSUE THREAD

**Issue by cgohla** [goldfirere/singletons#538]
Similar to #536.

When trying to generate `Show` instances for custom promoted types, as given in the [README](https://github.com/goldfirere/singletons#arrows-symbol-and-literals), using either 
```haskell
   decs2 <- showSingInstances [''Message]
```
or
```haskell
   decs2 <- singletons [d|
                deriving instance Show Message
                |]
```
fails with a compiler error
```
app/Main.hs:50:2: error:
    • Expected kind ‘PMessage’,
        but ‘z_a74u :: Message’ has kind ‘Message’
    • In the first argument of ‘SMessage’, namely ‘(z_a74u :: Message)’
      In the first argument of ‘Show’, namely
        ‘(SMessage (z_a74u :: Message))’
      In the stand-alone deriving instance for
        ‘Data.Singletons.ShowSing.ShowSing Text =>
         Show (SMessage (z_a74u :: Message))’
   |
50 | $(let customPromote :: Name -> Name
   |  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^...
```

Similar to #536, either method seems to disregard the custom name mapping.

**Comment by RyanGlScott** [comment]
This was fixed in #537. See the [`T536`](https://github.com/goldfirere/singletons/blob/6acdbf4baf2517df13515c6fb11b1a9d40cae3c3/singletons-base/tests/compile-and-dump/Singletons/T536.hs#L40) test case.

**Comment by cgohla** [comment]
> This was fixed in #537. See the [`T536`](https://github.com/goldfirere/singletons/blob/6acdbf4baf2517df13515c6fb11b1a9d40cae3c3/singletons-base/tests/compile-and-dump/Singletons/T536.hs#L40) test case.

Great. Thanks. 👍
